src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -665,31 +663,20 @@ sendMoney sendActions passphrase moneySource dstDistr = do
 getFullWalletHistory :: WalletWebMode m => CId Wal -> m ([CTx], Word)
 getFullWalletHistory cWalId = do
     addrs <- mapM decodeCIdOrFail =<< getWalletAddrs Ever cWalId
-    cHistory <- do
-        (mInit, cachedTxs) <- transCache <$> getHistoryCache cWalId
-
-        TxHistoryAnswer {..} <- untag @WalletSscType getTxHistory addrs mInit
-
-        -- Add allowed portion of result to cache
-        let fullHistory = taHistory <> cachedTxs
-            lenHistory = length taHistory
-            cached = drop (lenHistory - taCachedNum) taHistory
-        unless (null cached) $
-            updateHistoryCache
-                cWalId
-                taLastCachedHash
-                taCachedUtxo
-                (cached <> cachedTxs)
-
-        ctxs <- forM fullHistory $ addHistoryTx cWalId False
-        pure $ concatMap toList ctxs
+
+    blockHistory <- getHistoryCache cWalId >>= \case
+        Just hist -> pure $ DL.fromList hist
+        Nothing -> do
```

Comment:
@flyingleafe getting history, in this case, takes a loooooong time.
I don't mind not removing this code, but add logging msg plz